### DISCUSSION TITLE
Preventing the update of React from 16 to 18

### DISCUSSION THREAD

**Discussion by Thorbenl** [renovatebot/renovate#16423] [Request Help] [answered]
Hey :) 
We would like to stop updating react, or tell renovate to now include anything above `16.14.0` into the update PR. We have a few packages we want to pin. How do we go about this? :) 

Here is our current config:
```json
{
  "schedule": [
    "after 10pm every weekday",
    "before 5am every weekday",
    "every weekend"
  ],
  "masterIssue": true,
  "separateMajorMinor": false,
  "packageRules": [
   {
      "matchDepTypes": [
        "devDependencies"
      ],
      "matchManagers": [
        "npm"
      ],
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch",
        "pin",
        "digest",
        "lockFileMaintenance",
        "rollback",
        "bump"
      ],
      "groupName": "frontendDevDependencies",
      "groupSlug": "all-fr-devdeps",
      "semanticCommitType": "chore"
    },
    {
      "matchDepTypes": [
        "dependencies"
      ],
      "matchManagers": [
        "npm"
      ],
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch",
        "pin",
        "digest",
        "lockFileMaintenance",
        "rollback",
        "bump"
      ],
      "groupName": "frontendDependencies",
      "groupSlug": "all-fr-deps",
      "semanticCommitType": "fix"
    }
  ],
  "hostRules": [
   ... redacted
  ]
}
``` 

Our `package.json` has `"react": "^16.14.0",` but it includes it in our PR
![Screenshot 2022-07-05 at 08 22 38](https://user-images.githubusercontent.com/37654145/177262768-9373d5bf-6b92-47b3-8de1-8d2e38e6dc66.png)

Thanks!

**Comment by rarkins** [comment, marked as answer]
You could create a new package rule with:
- matchPackagePrefixes=["react"]
- matchCurrentVersion=^16.0.0
- allowedVersions=^16.0.0

Alternatively you could make it more complex and set dependencyDashboardApproval=true for major updates, so that you still see them.

You might also need a separate rule for react-dropzone if you only wnat to keep v11

**Reply by Thorbenl** [reply]
@rarkins Thanks for the rapid answer, appreciate it! :) I will mark this as resolved for now, and we will try out your solutions!
